coderd/userauth.go

```diff
@@ -604,6 +604,25 @@ func (api *API) userOAuth2Github(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// If we have a nil GitHub ID, that is a big problem. That would mean we link
+	// this user and all other users with this bug to the same uuid.
+	// We should instead throw an error. This should never occur in production.
+	//
+	// Verified that the lowest ID on GitHub is "1", so 0 should never occur.
+	if ghUser.GetID() == 0 {
+		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "The GitHub user returned an ID of 0, this should never happen. Please report this error.",
+			// If this happens, the User could either be:
+			//  - Empty, in which case all these fields would also be empty.
+			//  - Not a user, in which case the "Type" would be something other than "User"
+			Detail: fmt.Sprintf("Other user fields: name=%q, email=%q, type=%q",
+				ghUser.GetName(),
+				ghUser.GetEmail(),
+				ghUser.GetType(),
```

Comment:
Could this potentially leak another user's email? Would it be safer to log this instead?

Reply:
No it cannot. This comes from the authenticated request to github using your oauth token.

So this payload comes from:

- User does github oauth flow with coder
- Oauth flow succeeds
- Coder uses their oauth token to determine to make a new user or link to an existing
  - In this flow, the AuthentictedUser request fails from an `id=0`.

To leak someone else email, you would need to authenticate as that user via github.